apps/api/src/service/authService.ts

```diff
@@ -76,3 +82,46 @@ export async function isAssociatedRep(session: Partial<SessionData>, application
 
 	return false;
 }
+
+/**
+ * Handler to enforce Request is Authorized
+ * Insures Type safety for downstream Request Handler due to limitations with Express Type definitions
+ * @param request Express request
+ * @returns boolean
+ */
+export const isAuthenticatedRequest = (request: Request): request is AuthorizedRequest => {
+	return isRequestWithSession(request);
+};
+
+/**
+ * Validate User is allowed access to this specific Application based on userRole or userId
+ * @param request AuthorizedRequest with session, user and userId
+ * @returns boolean
+ */
+export async function canAccessRequest(
+	request: AuthorizedRequest,
+): AsyncResult<AuthorizedRequest, 'FORBIDDEN' | 'NOT_FOUND' | 'SYSTEM_ERROR'> {
+	const { session } = request;
+	const { user } = session;
+	const userRole = getUserRole(session);
+	const requestedId = request.params.applicationId || request.body.applicationId;
+
+	const hasSpecialAccess =
+		userRole === userRoleSchema.Values.DAC_MEMBER ||
+		userRole === userRoleSchema.Values.DAC_CHAIR ||
+		(await isAssociatedRep(session, requestedId));
+
+	const result = await getApplicationById({ applicationId: requestedId });
+	if (result.success) {
+		const { data } = result;
+		const { userId } = user;
+		const canAccess = data.userId === userId || hasSpecialAccess;
+
+		if (!canAccess) {
+			return failure('FORBIDDEN', 'User does not have permission to access or modify this application.');
+		}
+	} else {
+		return result;
+	}
+	return success(request);
```

Comment:
I think we can just return a `Result<void>` since there is no new information returned on success, only error codes on failure. This is the equivalent of returning "success OR some error code".